core/src/main/java/io/lionweb/lioncore/java/serialization/JsonSerialization.java

```diff
@@ -414,30 +414,13 @@ private List<SerializedClassifierInstance> sortLeavesFirst(
     while (sortedList.size() < originalList.size()) {
       int initialLength = sortedList.size();
       for (int i = 0; i < nodesToSort.size(); i++) {
-        SerializedClassifierInstance n = nodesToSort.get(i);
-        if (n instanceof SerializedNodeInstance) {
-          SerializedNodeInstance serializedNodeInstance = (SerializedNodeInstance) n;
-          if (serializedNodeInstance.getParentNodeID() == null
-              || sortedList.stream()
-                  .anyMatch(
-                      sn -> Objects.equals(sn.getID(), serializedNodeInstance.getParentNodeID()))) {
-            sortedList.add(n);
-            nodesToSort.remove(i);
-            i--;
-          }
-        } else if (n instanceof SerializedAnnotationInstance) {
-          SerializedAnnotationInstance serializedAnnotationInstance =
-              (SerializedAnnotationInstance) n;
-          if (serializedAnnotationInstance.getParentNodeID() == null
-              || sortedList.stream()
-                  .anyMatch(
-                      sn ->
-                          Objects.equals(
-                              sn.getID(), serializedAnnotationInstance.getParentNodeID()))) {
-            sortedList.add(n);
-            nodesToSort.remove(i);
-            i--;
-          }
+        SerializedClassifierInstance node = nodesToSort.get(i);
```

Comment:
should we call the variable `classifierInstance` or `instance` or `ci`?

Reply:
I'd rather rename the `SerializedClassifierInstance` to `SerializedNode` (and so `node` &rarr; `serializedNode`), actually.

Reply:
I am confused about the term "node", as it could mean both node as in proper member of the AST (so no annotation instances) and node as in proper member of the AST or annotation instance. I think it would be useful to have separate terms for these two concepts

Reply:
In TS and C# code I tend to use the names `node` and `serializedNode` – can we do that here as well?

Reply:
I am not super-convinced about this point, but perhaps we can revisit it later and keep things as they are for the sake of getting this PR over the line and the release done quick;y

Reply:
Marking this as resolved